blog/2025/issue-credentials-over-openid4vci.adoc

```diff
@@ -0,0 +1,358 @@
+
+These formats allow issuers to provide portable, verifiable digital proofs that support selective disclosure and offline verification in decentralized identity ecosystems.
+
+*Keycloak currently supports both SD-JWT VC and JWT VC for issuance*, while mDL/mdoc listed above is part of the OpenID4VCI specification.
+
+=== Concrete Use-Cases Enabled by OpenID4VCI
+
+Verifiable credentials open up a variety of real-world applications. Examples include:
+
+- Governments issuing digital identity cards or driver’s licenses that citizens can present when booking hotels, opening bank accounts, or accessing public services, while revealing only the information necessary for the transaction (e.g., age or residency) without exposing full personal details.
+- City councils issuing verifiable birth certificates that universities and hospitals can validate without a central lookup.
+- Universities issuing digital diplomas that employers can instantly verify for authenticity.
+- Companies issuing employee badges as verifiable credentials for office access or remote authentication.
+- Event organizers issuing verifiable tickets that can be validated offline.
+- Professional associations issuing membership credentials such as in our OAuth SIG example for verifying access to gated resources or conference venues.
+- Digital movie or event tickets that are tamper-evident and can be verified offline.
+
+These examples illustrate the variety of situations where verifiable credentials remove the need for direct communication between verifier and issuer while preserving trust.
+
+=== Key Flows in OpenID4VCI
+
+OpenID4VCI defines two primary flows for credential issuance: the **Authorization Code Flow** and the **Pre-Authorized Code Flow**. These flows determine how a wallet (holder’s application) obtains an access token to request a VC from the issuer.
+
+==== Authorization Code Flow
+
+The Authorization Code Flow is **interactive** and requires the holder to authenticate and consent at the issuer’s authorization endpoint. It is ideal for scenarios where explicit user approval is required, or additional claims must be collected.
+
+**Steps:**
+
+1. The **Wallet** sends an Authorization Request to the **Issuer’s Authorization Endpoint**, requesting credentials.
+2. The **Issuer** authenticates the **User** and requests consent for the issuance.
+3. The **User** provides credentials and grants consent.
+4. The **Issuer** returns an **Authorization Code** to the Wallet.
+5. The **Wallet** exchanges the code at the **Issuer Token Endpoint** for an **Access Token**.
+6. The **Wallet** requests the **Verifiable Credential** from the **Issuer Credential Endpoint** using the Access Token.
+7. The **Issuer** returns the **VC** to the Wallet.
+
+.Credential issuance via Authorization Code Flow
+image::${blogImages}/openid4vci/authorization-code-flow.png[alt="Credential issuance via Authorization Code Flow", align=center, width=800]
```

Comment:
In this image, to ensure consistency between a request and its response consistent, it might be good add
`Receive Access Token` as a response
after
`Exchange Authorization Code for Access Token (Token Endpont)` as a request